stats/opentelemetry/internal/tracing/custom_carrier_test.go

```
@@ -64,27 +83,48 @@ func (s) TestGet(t *testing.T) {
 			key:  "key1",
 			want: "",
 		},
+		{
+			name: "grpc-trace-bin key",
+			md:   metadata.MD{},
+			key:  "grpc-trace-bin",
+			want: base64.StdEncoding.EncodeToString([]byte{0x01, 0x02, 0x03}),
+		},
 		{
 			name: "non-grpc-trace-bin key",
-			md:   metadata.Pairs("non-trace-bin", "\x01\x02\x03"),
+			md:   metadata.Pairs("non-trace-bin", base64.StdEncoding.EncodeToString([]byte{0x01, 0x02, 0x03})),
 			key:  "non-trace-bin",
-			want: "",
+			want: "", // -bin key that isn't grpc-trace-bin should always get empty string
 		},
 	}
 
 	for _, test := range tests {
-		ctx, cancel := context.WithCancel(context.Background())
 		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			if strings.HasSuffix(test.key, "-bin") { // for binary headers set `grpc-trace-bin` using `stats.SetIncomingTrace()`
+				b, _ := base64.StdEncoding.DecodeString(test.want)
```

Same here, don't ignore errors.

---

Done